src/ArtGobblers.sol

```diff
@@ -786,6 +794,76 @@ contract ArtGobblers is GobblersERC721, LogisticVRGDA, Owned, ERC1155TokenReceiv
+    /// @param amount The amount of goo to approve spender for.
+    function approveGoo(address spender, uint256 amount) public virtual returns (bool) {
+        gooAllowance[msg.sender][spender] = amount;
+
+        emit Approval(msg.sender, spender, amount);
+
+        return true;
+    }
+
+    /// @notice Transfer goo directly from your emission balance,
+    /// to a recipient's emission balance.
+    /// @param to The goo recipient's address.
+    /// @param gooAmount The amount of goo to transfer.
+    function transferGoo(address to, uint256 gooAmount) external returns (bool) {
+        // Will revert due to underflow if we're decreasing by more than the user's current balance.
+        uint256 fromUpdatedBalance = gooBalance(msg.sender) - gooAmount;
+        getUserData[msg.sender].lastBalance = uint128(fromUpdatedBalance);
+        // Don't need to do checked addition in the increase case.
+        uint256 toUpdatedBalance;
+        unchecked {
+            toUpdatedBalance = gooBalance(to) + gooAmount;
+            getUserData[to].lastBalance = uint128(toUpdatedBalance);
+        }
+
+        // Update both accounts lastTimestamp now
+        getUserData[msg.sender].lastTimestamp = uint64(block.timestamp);
+        getUserData[to].lastTimestamp = uint64(block.timestamp);
+
+        emit GooBalanceUpdated(msg.sender, fromUpdatedBalance);
+        emit GooBalanceUpdated(to, toUpdatedBalance);
+        emit GooTransfer(msg.sender, to, gooAmount);
+    }
+
+    /// @notice Transfer goo directly from a user's emission balance,
+    /// to a recipient's emission balance.
+    /// @param from The goo sender's address.
+    /// @param to The goo recipient's address.
+    /// @param gooAmount The amount of goo to transfer.
+    function transferGooFrom(
+        address from,
+        address to,
+        uint256 gooAmount
+    ) external returns (bool) {
+        uint256 allowed = gooAllowance[from][msg.sender]; // Saves gas for limited approvals.
+
+        if (allowed != type(uint256).max) gooAllowance[from][msg.sender] = allowed - gooAmount;
+
+        // Will revert due to underflow if we're decreasing by more than the user's current balance.
+        uint256 fromUpdatedBalance = gooBalance(from) - gooAmount;
+        getUserData[from].lastBalance = uint128(fromUpdatedBalance);
+        // Don't need to do checked addition in the increase case.
+        uint256 toUpdatedBalance;
+        unchecked {
+            toUpdatedBalance = gooBalance(to) + gooAmount;
+            getUserData[to].lastBalance = uint128(toUpdatedBalance);
+        }
+
+        // Update both accounts lastTimestamp now
+        getUserData[from].lastTimestamp = uint64(block.timestamp);
+        getUserData[to].lastTimestamp = uint64(block.timestamp);
+
+        emit GooBalanceUpdated(from, fromUpdatedBalance);
+        emit GooBalanceUpdated(to, toUpdatedBalance);
+        emit GooTransfer(from, to, gooAmount);
```

Comment:
What's the stance on events here? Keep all three, or just use GooTransfer (which seems to convey all the same properties). On the other hand, supporting older events might be better for frontend integration.